packages/template-retail-react-app/app/components/sf-payments-express-buttons/index.jsx

```diff
@@ -0,0 +1,827 @@
+            }
+
+            const handlePaymentMethodsRendered = (details) => {
+                if (onPaymentMethodsRendered && details.detail.rendered.length > 0) {
+                    onPaymentMethodsRendered()
+                }
+            }
+
+            const paymentMethodSet = {
+                paymentMethods: paymentConfig.paymentMethods,
+                paymentMethodSetAccounts: paymentConfig.paymentMethodSetAccounts
+            }
+
+            const config = {
+                theme: buildTheme({
+                    expressButtonLayout,
+                    ...(usage === EXPRESS_BUY_NOW && {
+                        expressButtonLabels: {
+                            applepay: 'buy',
+                            googlepay: 'buy',
+                            paypal: 'buynow',
+                            venmo: 'buynow'
+                        }
+                    })
+                }),
+                actions: {
+                    onClick: onClick,
+                    onShippingAddressChange: onShippingAddressChange,
+                    onShippingMethodChange: onShippingMethodChange,
+                    createIntent: createIntentFunction,
+                    onPayerApprove: onPayerApprove
+                },
+                options: {
+                    shippingAddressRequired: true,
+                    emailAddressRequired: true,
+                    billingAddressRequired: true,
+                    phoneNumberRequired: true,
+                    useManualCapture: !cardCaptureAutomatic,
+                    maximumButtonCount
+                }
+            }
+
+            const paymentRequest = {
+                amount: initialAmount,
+                currency: paymentCurrency,
+                country: 'US', // TODO: see W-18812582
+                locale: intl.locale
+            }
+
+            containerElementRef.current.innerHTML = '<div></div>'
+
+            const paymentElement = containerElementRef.current.firstChild
+
+            paymentElement.addEventListener('sfp:paymentcancel', onCancel)
+            paymentElement.addEventListener('sfp:paymentapprove', onApproveEvent)
+            paymentElement.addEventListener('sfp:paymenterror', paymentError)
+            paymentElement.addEventListener(
+                'sfp:paymentmethodsrendered',
+                handlePaymentMethodsRendered
+            )
+
+            expressComponent.current = sfp.express(
+                metadata,
+                paymentMethodSet,
+                config,
+                paymentRequest,
+                paymentElement,
+                usage
+            )
+        }
+
+        // Cleanup on unmount
+        return () => {
+            expressComponent.current?.destroy()
+            expressComponent.current = null
+        }
+    }, [sfp, metadata, paymentConfig, cardCaptureAutomatic, containerElementRef.current])
+
+    return <Box ref={containerElementRef} data-testid={'sf-payments-express'} />
```

Comment:
There is a lot of code in this file. Could it be refactored into smaller files/hooks/functions etc? It's hard to know at a glance how all the different pieces work together.